server/pulp/server/db/model/__init__.py

```diff
@@ -49,6 +52,25 @@ def __init__(self, *args, **kwargs):
 
     meta = {'abstract': True}
 
+    def clean(self):
```

Comment:
`clean()` is called by mongoengine as [part of the save process](http://docs.mongoengine.org/en/latest/guide/document-instances.html?highlight=clean#pre-save-data-validation-and-cleaning).

The point about this raising an exception when an instance is saved when the issue is actually on the document definition is well stated and correct. I still decided to do it this way for 3 reasons:

1. We lack a mechanism to validate that "all pulp mongoengine based models" validate in a specific way. Even putting it here is a poor attempt to address that problem because we can't be guaranteed that all pulp mongoengine models will even inherit from `AutoRetryDocument`.

2. As developers and testers, we will see the errors before actual users do so it will have the intended effect.

3. We get to delete this once we handle [issue 625](https://pulp.plan.io/issues/625).

This is a quick-and-dirty way for us to get some confidence that subclasses defined the right fields. Currently there is only 1 global field that should be defined, `_ns`.